src/components/report-viewer/report-viewer.tsx

```diff
@@ -109,34 +111,51 @@ export default function ReportViewer({
         [reportTreeMap]
     );
 
+    // Ref for resizing
+    const panelGroupRef = useRef<ImperativePanelGroupHandle>(null);
+
+    // sizes in percent
+    const LEFT_PANEL_MIN_SIZE = 15;
+    const LEFT_PANEL_DEFAULT_SIZE = 25;
+    const RIGHT_PANEL_MIN_SIZE = 50;
+
     return (
-        <Grid
-            container
-            ref={handleReportVerticalPositionFromTop}
-            sx={{
-                height: reportContainerHeight,
-            }}
-        >
-            <Grid item sm={3} sx={{ borderRight: (theme) => `1px solid ${theme.palette.divider}` }}>
-                <VirtualizedTreeview
-                    expandedTreeReports={expandedTreeReports}
-                    setExpandedTreeReports={setExpandedTreeReports}
-                    selectedReportId={selectedReport.id}
-                    reportTree={reportTree}
-                    onSelectedItem={handleSelectedItem}
-                    highlightedReportId={highlightedReportId}
-                />
-            </Grid>
-            <Grid item xs={12} sm={9}>
-                <LogTable
-                    selectedReport={selectedReport}
-                    reportType={reportType}
-                    severities={severities}
-                    onRowClick={onLogRowClick}
-                    onFiltersChanged={onFiltersChanged}
-                    resetFilters={resetFilters}
-                />
-            </Grid>
-        </Grid>
+        <Box width={'100%'} ref={handleReportVerticalPositionFromTop} sx={{ height: reportContainerHeight }}>
+            <PanelGroup direction="horizontal" ref={panelGroupRef}>
+                <Panel id="treeview-panel" minSize={LEFT_PANEL_MIN_SIZE} defaultSize={LEFT_PANEL_DEFAULT_SIZE}>
+                    <VirtualizedTreeview
+                        expandedTreeReports={expandedTreeReports}
+                        setExpandedTreeReports={setExpandedTreeReports}
+                        selectedReportId={selectedReport.id}
+                        reportTree={reportTree}
+                        onSelectedItem={handleSelectedItem}
+                        highlightedReportId={highlightedReportId}
+                    />
+                </Panel>
+                <Box
+                    display="flex"
+                    alignItems="center"
+                    sx={{
+                        height: '100%',
+                        borderLeft: (theme) => `1px solid ${theme.palette.divider}`,
+                        borderRight: (theme) => `1px solid ${theme.palette.divider}`,
+                    }}
```

Comment:
```suggestion
                    const styles = {
    panelHandlerContainer: (theme: Theme) => ({
        borderRight: `1px solid ${theme.palette.divider}`,
        marginRight: theme.spacing(1),
    }),
};
```

Can you extract the style as we do modt of the time ? And I propose this, it's a bit different, but just a proposition